File: 122-best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        total_prof = 0 
        for i in range(1,len(prices)):
            if prices[i-1]< prices[i]:
                total_prof += (prices[i]-prices[i-1])

        return total_prof
        

File: 122-best-time-to-buy-and-sell-stock-ii/best-time-to-buy-and-sell-stock-ii.java
class Solution {
    public int maxProfit(int[] nums) {
        int profit = 0;
        
        int x = 0;

        for(int i = 0;i<nums.length;i++){

            if(nums[i]>nums[x]){
                profit += (nums[i] -nums[x]); 
            }
            x = i;
        }

        return profit;
        
    }
}